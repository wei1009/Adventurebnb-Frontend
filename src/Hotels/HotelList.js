import React from "react";
import '../CSS/HoteList.css';

/** Show information about hotels by city or zip code
 *
 * Is rendered by Hotel to show a "card" for each hotel.
 *
 * Hotel -> HotelList
 */

function HotelList(
    { hotelName, hotelFeature, street, city, state, postalCode, web, phone, image, rate, ranking,
        checkInDate, checkOutDate, hotelCode, adult, children }) {
        
    let absoulteUrl;



    return (
        <>
            <div className="container hotelList">
                <div className="row">
                    <div className="col-lg-4 hotel-list-image" style={{ backgroundImage: `url(${image})`}}>
                    </div>

                    <div className=" col-lg-6 ">
                        <h4 >
                            {hotelName}
                        </h4>
                        <div className="hotelFeature">
                            <h6 className="hotelFeatureContent"> {hotelFeature}</h6>
                        </div>
                        <div className=" hotelInfomation">
                            <div >{street}, {city}, {state} {postalCode}</div>
                            <a href={web} target="_blank">{web}</a>
                            <div>{phone}</div>
                        </div>
                    </div>
                    <div className="hotelSubInformation col-lg-2">
                        <div className="ranking">
                            <div>
                                rate:{rate}
                            </div>
                            <div>
                                ranking:{ranking}
                            </div>
                        </div>
                        <a className="btn btn-md btn-primary hotelListBtn"
                            href={`/hotel?type=hotel&hotel_code=${hotelCode}&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=${adult}&children=${children}`}>
                            Go to hotel
                        </a>
                    </div>
                </div>
            </div>
        </>
    )
};

export default HotelList;