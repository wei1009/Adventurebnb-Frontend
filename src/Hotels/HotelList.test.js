import React from "react";
import { render } from "@testing-library/react";
import HotelList from "./HotelList";
import { MemoryRouter } from "react-router";

// TODO: woefully under-tested!

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <HotelList 
          hotelName ='test'
          image = 'https://photos.hotelbeds.com/giata/41/412145/412145a_hb_a_001.jpg' 
          street ='test street'
          city ='test'
          state = 'test'
          postalCode =' 12345'
          hotelFeature = 'test'
          web = '/'
          checkInDate = '2022-03-10'
          checkOutDate = '2022-03-12'
          hotelCode = '12345'
          adult ='2'
          children='3'
          />
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});