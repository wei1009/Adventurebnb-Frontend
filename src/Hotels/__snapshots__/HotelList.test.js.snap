// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container hotelList"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-4 hotel-list-image"
        style="background-image: url(https://photos.hotelbeds.com/giata/41/412145/412145a_hb_a_001.jpg);"
      />
      <div
        class=" col-lg-6 "
      >
        <h4>
          test
        </h4>
        <div
          class="hotelFeature"
        >
          <h6
            class="hotelFeatureContent"
          >
             test
          </h6>
        </div>
        <div
          class=" hotelInfomation"
        >
          <div>
            test street, test, test  12345
          </div>
          <a
            href="/"
          >
            /
          </a>
          <div />
        </div>
      </div>
      <div
        class="hotelSubInformation col-lg-2"
      >
        <div
          class="ranking"
        >
          <div>
            rate:
          </div>
          <div>
            ranking:
          </div>
        </div>
        <a
          class="btn btn-md btn-primary hotelListBtn"
          href="http://localhost:3000/hotel?type=hotel&hotel_code=12345&checkInDate=2022-03-10&checkOutDate=2022-03-12&adult=2&children=3"
        >
          Go to hotel
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
